src/project.coffee

```diff
@@ -203,10 +204,12 @@ class Project extends Model
     for provider in @directoryProviders
       break if directory = provider.directoryForURISync?(projectPath)
     if directory is null
-      # This should never happen because DefaultDirectoryProvider should always
-      # return a Directory.
-      throw new Error(projectPath + ' could not be resolved to a directory')
-    @rootDirectories.push(directory)
+      # DefaultDirectoryProvider doesn't handle specific directory protocols
+      # Atom Packages may use packageDirectoryPaths to add thier own paths later.
```

Comment:
No, that's for the case when you have a directory sitting on your server and have a directory provider that serves files from it to your Atom instance (in your package) - e.g. http://someserver:port/workspace_path
When you Atom restarts and tries to load this directory, it will use its default-directory-provider and hence, fail and destroy your project uri - or path, because the `default-directory-provider` and the default pathwatcher's `Directory` normalizes the given path (which could sometimes not be a filesystem path, but a uri).

So, this diff is to make the `default-directory-provider` skip loading a certain path, if it can't (doesn't exist on the local filesystem).